apps/api/src/server/appeals/appeals/appeals.service.js

```diff
@@ -0,0 +1,124 @@
+import { add, addBusinessDays, isAfter, isBefore, sub } from 'date-fns';
+import appealRepository from '../../repositories/appeal.repository.js';
+import config from '../config.js';
+import {
+	BANK_HOLIDAY_FEED_DIVISION_ENGLAND,
+	BANK_HOLIDAY_FEED_URL,
+	ERROR_NOT_FOUND
+} from '../constants.js';
+
+/**
+ * @typedef {import('@pins/api').Appeals.BankHolidayFeedEvents} BankHolidayFeedEvents
+ * @typedef {import('@pins/api').Appeals.TimetableDeadlineDate} TimetableDeadlineDate
+ */
+
+/**
+ * Count the number of bank holidays between two dates
+ *
+ * @param {Date} dateFrom
+ * @param {Date} dateTo
+ * @param {BankHolidayFeedEvents} bankHolidays
+ * @returns {number}
+ */
+const bankHolidaysBetweenDates = (dateFrom, dateTo, bankHolidays) =>
+	bankHolidays.filter(
+		({ date }) =>
+			isAfter(new Date(date), sub(new Date(dateFrom), { days: 1 })) &&
+			isBefore(new Date(date), add(new Date(dateTo), { days: 1 }))
+	).length;
+
+/**
+ * @param {string} division
+ * @returns {Promise<BankHolidayFeedEvents>}
+ */
+const fetchBankHolidaysForDivision = async (division = BANK_HOLIDAY_FEED_DIVISION_ENGLAND) => {
+	const bankHolidayFeed = await fetch(BANK_HOLIDAY_FEED_URL);
```

Comment:
For the caching, we could cache the response but seems pretty responsive and it’s not an action that will get used constantly so perhaps not worth it right now and perhaps something we can keep in mind in case we should in the future.

There is some error handling on the request in case it fails.